src/Orleans.Serialization/Buffers/Writer.cs

```diff
@@ -415,7 +431,7 @@ public void WriteVarUInt32(uint value)
             // Since this method writes a ulong worth of bytes unconditionally, ensure that there is sufficient space.
             EnsureContiguous(sizeof(ulong));
 
-            var pos = _bufferPos;
+            nuint pos = (uint)_bufferPos;
             var neededBytes = BitOperations.Log2(value) / 7;
```

Comment:
```suggestion
            var neededBytes = (int)((uint)BitOperations.Log2(value) / 7);
```
More ugly code, but [shorter machine code](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgDkBXfGKASzFwBuGjWIBmJqQYBhBgG8aDJUwm8AdhgYBZcgAoO6zQDdsAGw4wAlAwC8APgYAhXhgDyABx7YMvCGtwsADIQAOakuibmVgwA9AwA7MLUyioMhtrhBhoMkRbW9gy6hpa6+sXObp5Q3r7+QaHhudFx8ZZJAL5AA==).

Reply:
Nice work. I wonder if this results in measurable differences.
There are serialization microbenchmarks in the test/Benchmarks directory which can be invoked with
`dotnet run -c Release -f net7.0 -- suite --filter *erializeBenchmark` to run the serialization comparison benchmarks.

Reply:
Did two runs of the benchmarks (current main, then changed to the uint-division). The runs aren't comparable 1:1, but judging based on _Ratio_-column it improved a bit, which I expected because of less code and the _mul_ instead of _imul_.

#### Current main branch

```
|            Method |       Mean |    Error |   StdDev | Ratio | RatioSD |   Gen0 | Payload | Allocated | Alloc Ratio |
|------------------ |-----------:|---------:|---------:|------:|--------:|-------:|--------:|----------:|------------:|
|           Orleans |   107.1 ns |  2.12 ns |  3.17 ns |  1.00 |    0.00 |      - |    20 B |         - |          NA |
|          Utf8Json |   169.6 ns |  3.23 ns |  7.50 ns |  1.59 |    0.09 |      - |   154 B |         - |          NA |
|    SystemTextJson |   571.5 ns |  5.31 ns |  4.97 ns |  5.29 |    0.14 | 0.0019 |   154 B |      56 B |          NA |
| MessagePackCSharp |   115.0 ns |  0.84 ns |  0.71 ns |  1.06 |    0.03 | 0.0014 |    10 B |      40 B |          NA |
|       ProtobufNet |   283.3 ns |  3.69 ns |  2.88 ns |  2.61 |    0.06 |      - |    18 B |         - |          NA |
|          Hyperion |   165.0 ns |  3.33 ns |  7.66 ns |  1.53 |    0.06 | 0.0019 |    39 B |      56 B |          NA |
|    NewtonsoftJson | 1,700.0 ns | 33.79 ns | 78.31 ns | 16.21 |    0.87 | 0.0763 |   154 B |    2080 B |          NA |
|          SpanJson |   192.5 ns |  2.09 ns |  2.64 ns |  1.79 |    0.05 | 0.0067 |   154 B |     184 B |          NA |
```

#### With unsigned divsion

```
|            Method |        Mean |     Error |    StdDev | Ratio | RatioSD |   Gen0 | Payload | Allocated | Alloc Ratio |
|------------------ |------------:|----------:|----------:|------:|--------:|-------:|--------:|----------:|------------:|
|           Orleans |    95.51 ns |  0.351 ns |  0.311 ns |  1.00 |    0.00 |      - |    20 B |         - |          NA |
|          Utf8Json |   156.21 ns |  1.141 ns |  0.953 ns |  1.64 |    0.01 |      - |   154 B |         - |          NA |
|    SystemTextJson |   537.95 ns |  4.033 ns |  3.575 ns |  5.63 |    0.04 | 0.0019 |   154 B |      56 B |          NA |
| MessagePackCSharp |   118.38 ns |  2.122 ns |  1.985 ns |  1.24 |    0.02 | 0.0014 |    10 B |      40 B |          NA |
|       ProtobufNet |   273.29 ns |  1.479 ns |  1.311 ns |  2.86 |    0.02 |      - |    18 B |         - |          NA |
|          Hyperion |   152.09 ns |  1.320 ns |  1.170 ns |  1.59 |    0.01 | 0.0019 |    39 B |      56 B |          NA |
|    NewtonsoftJson | 1,594.22 ns | 15.754 ns | 14.736 ns | 16.70 |    0.17 | 0.0763 |   154 B |    2080 B |          NA |
|          SpanJson |   189.82 ns |  1.633 ns |  1.528 ns |  1.99 |    0.02 | 0.0067 |   154 B |     184 B |          NA |
```

Full results:
[Benchmarks.zip](https://github.com/dotnet/orleans/files/9890683/Benchmarks.zip)